source/components/inputs/input.ts

```diff
@@ -62,4 +69,14 @@ export class InputComponent<T> implements AfterViewInit, OnInit {
 			this.change.emit(value);
 		}
 	}
+
+	showLabel(): string {
+		return this.labelState = slide.show;
+	}
+
+	hideLabel(): string {
+		return this.object.isNullOrEmpty(this.value)
```

Comment:
And here this would just be `!this.value`, or `this.value` if you reverse the two cases below